@here/harp-mapview/lib/geometry/PhasedTileGeometryLoader.ts

```diff
@@ -0,0 +1,420 @@
+
+                    // Check if that technique fits the expected kindToCreate.
+                    return techniqueKinds.has(kindToCreate);
+                } else {
+                    return techniqueKind === kindToCreate;
+                }
+            };
+
+            geometryCreator.createObjects(tile, decodedTile, filter);
+
+            const textFilter = (technique: Technique): boolean => {
+                if (
+                    !isPoiTechnique(technique) &&
+                    !isLineMarkerTechnique(technique) &&
+                    !isTextTechnique(technique)
+                ) {
+                    return false;
+                }
+                return filter(technique);
+            };
+
+            // TextElements do not get their geometry created by Tile, but are managed on a
+            // higher level.
+            geometryCreator.createTextElements(tile, decodedTile, textFilter);
+
+            geometryCreator.preparePois(tile, decodedTile);
+        }
+    }
+
+    protected processTechniques(
+        enabledKinds: GeometryKindSet | undefined,
+        disabledKinds: GeometryKindSet | undefined
+    ): void {
+        const decodedTile = this.m_decodedTile;
+
+        if (decodedTile === undefined) {
+            return;
+        }
+
+        for (const technique of decodedTile.techniques) {
+            // Make sure that all technique have their geometryKind set, either from the Theme or
+            // their default value.
+            let geometryKind = technique.kind;
+
+            // Set default kind based on technique.
+            if (geometryKind === undefined) {
+                if (isFillTechnique(technique)) {
+                    geometryKind = GeometryKind.Area;
+                } else if (
+                    isLineTechnique(technique) ||
+                    isDashedLineTechnique(technique) ||
+                    isSolidLineTechnique(technique) ||
+                    isSegmentsTechnique(technique) ||
+                    isExtrudedLineTechnique(technique)
+                ) {
+                    geometryKind = GeometryKind.Line;
+                } else if (isExtrudedPolygonTechnique(technique)) {
+                    geometryKind = GeometryKind.Building;
+                } else if (
+                    isPoiTechnique(technique) ||
+                    isLineMarkerTechnique(technique) ||
+                    isTextTechnique(technique)
+                ) {
+                    geometryKind = GeometryKind.Label;
+                } else {
+                    geometryKind = GeometryKind.All;
+                }
+
+                technique.kind = geometryKind;
+            }
+        }
+
+        // Speedup and simplify following code: Test all techniques if they intersect with
+        // enabledKinds and disabledKinds, in which case they are flagged. The disabledKinds can be
+        // ignored hereafter.
+        const geometryCreator = new TileGeometryCreator();
```

Comment:
Same as above, use singleton instance instead of creating new objects?

Reply:
I am a bit torn here. Should it be a singleton, or owned by MapView... I can live with the current way of handling it.

Reply:
As currently implemented I think you could also make all the methods static in TileGeometryCreator. Unless there is a good reason better to prevent the unnecessary allocations.

Reply:
Implemented as a singleton.